sklearn/neighbors/base.py

```diff
@@ -201,23 +207,30 @@ def _fit(self, X):
             self._fit_method = 'kd_tree'
             return self
 
-        X = check_array(X, accept_sparse='csr')
+        X = check_array(X, accept_sparse='csr',
+                        force_all_finite=kill_missing)
 
         n_samples = X.shape[0]
         if n_samples == 0:
             raise ValueError("n_samples must be greater than 0")
 
         if issparse(X):
-            if self.algorithm not in ('auto', 'brute'):
-                warnings.warn("cannot use tree with sparse input: "
-                              "using brute force")
-            if self.effective_metric_ not in VALID_METRICS_SPARSE['brute']:
-                raise ValueError("metric '%s' not valid for sparse input"
-                                 % self.effective_metric_)
-            self._fit_X = X.copy()
-            self._tree = None
-            self._fit_method = 'brute'
-            return self
+            if not kill_missing:
+                raise ValueError(
+                    "Nearest neighbor algorithm does not currently support"
+                    "the use of sparse matrices."
+                )
+            else:
```

Comment:
Please correct me if I am mistaken, but it seems the two "if" statements following the "else" are not mutually exclusive. This would preclude the use of two "elif"s instead right? However, I think I can simply remove the "else" as it seems to be redundant there.